jupyter/jupyter-book
# Wrap notebook HTML in {% raw %} tag

Right now, notebook HTML will be processed through Jekyll. This causes
problems for notebooks with complicated LaTeX – for example, [this page
from the Data 100 textbook][1] breaks since it has the substring `{{` in
the LaTeX.

Notebook authors shouldn't have to worry about messing up their book by
accidentally including a reserved Jekyll/Liquid keyword in their
notebook, so this commit wraps the notebook's generated HTML in a
`{% raw %}` tag.

Technically, notebook authors won't be able to have the string
`{% endraw %}` anywhere within their notebooks, but this is probably
less likely to appear than double braces.

The `citations.ipynb` file is also updated to remove uses of
`{% endraw %}`.

To test this, I did the following:

1. `pip install -e .`
2. Within the `book_template/` folder, run `make book` and verify that
   the content in `_build/features/notebooks.html` is wrapped with a `raw` tag.
3. Run `jekyll serve` and verify that
   http://127.0.0.1:4001/jupyter-book/features/notebooks.html renders properly.

[1]: https://www.textbook.ds100.org/ch/15/bias_modeling.html
